app/uk/gov/nationalarchives/omega/editorial/services/jms/ResponseBuilder.scala

```diff
@@ -44,26 +44,50 @@ class ResponseBuilder[F[_] : ResponseBuilder.ME : Logger] {
 
   def createResponseText(jmsMessage: JmsMessage): F[String] =
     jmsMessage.getStringProperty(sidHeaderKey) match {
-      case Some(sidValue) if sidValue == sid1 =>
-        logger.info(s"got a message with sid header '$sidValue', trying to parse payload") *>
-          getEditSet(jmsMessage).flatMap { editSet =>
-            logger.info("parsed payload, creating edit set response") *>
-              me.pure(Json.toJson(editSet).toString)
-          }
-
+      case Some(sidValue) if sidValue == SID.GetEditSet =>
+        handleRequestForGettingAnEditSet(jmsMessage, sidValue)
+      case Some(sidValue) if sidValue == ResponseBuilder.SID.GetEditSetRecord =>
+        handleRequestForGettingAnEditSetRecord(jmsMessage)
       case unknown =>
-        logger.error(s"Expected a SID value of $sid1, but got $unknown, failing") *>
-          me.raiseError(new NotImplementedError(s"unhandled case for SID $unknown"))
-
+        onUnhandledCase(s"SID is unrecognised: [$unknown]")
     }
 
+  private def handleRequestForGettingAnEditSet(jmsMessage: JmsMessage, sidValue: String): F[String] =
+    logger.info(s"got a message with sid header '$sidValue', trying to parse payload") *>
+      getEditSet(jmsMessage).flatMap { editSet =>
+        logger.info("parsed payload, creating edit set response") *>
+          me.pure(Json.toJson(editSet).toString)
+      }
+
+  private def handleRequestForGettingAnEditSetRecord(jmsMessage: JmsMessage): F[String] =
+    asGetEditSetRecordRequest(jmsMessage).flatMap(getEditSetRecordRequest =>
+      getEditSetRecordByOCI(getEditSetRecordRequest.recordOci)
+        .map(editSetRecord => me.pure(Json.toJson(editSetRecord).toString))
+        .getOrElse(
+          onUnhandledCase(
+            s"Unable to find record for Edit Set with OCI [${getEditSetRecordRequest.editSetOci}] and Record OCI [${getEditSetRecordRequest.recordOci}]"
+          )
+        )
+    )
+
+  private def onUnhandledCase(errorMessage: String): F[String] =
+    logger.error(errorMessage) *> me.raiseError(
+      new NotImplementedError(s"We don't yet handle this case: $errorMessage")
+    )
+
   private def getEditSet(jmsMessage: JmsMessage): F[EditSet] =
     for {
       messageText <- messageText(jmsMessage)
       // TODO We'll need to get the OCI from the GetEditSet eventuly
       _ <- parse[GetEditSet](messageText)
     } yield editSets.editSet1
 
+  private def asGetEditSetRecordRequest(jmsMessage: JmsMessage): F[GetEditSetRecord] =
```

Comment:
Not an issue but this could be written using flatMap:
```
messageText(jmsMessage).flatMap(parse[GetEditSetRecord])
```

Reply:
I like it.

Reply:
Changed.

Reply:
Marking resolved, as you've approved, @felixgb .